### ISSUE TITLE
Allow setting subPath for Kubernetes VolumeMounts in CloudDriver

### ISSUE THREAD

**Issue by RoryQ** [spinnaker/spinnaker#2082]
I'm deploying some services to Kubernetes and want to mount a single config.json file in the root of my app folder. When deploying to kubernetes manually I can specify a subPath and a full mountPath to get this behaviour, but with spinnaker setting a subPath in the deployment json isn't getting passed through to kubernetes ([see this issue ](https://github.com/kubernetes/kubernetes/issues/44815#issuecomment-297077509))

Deployment json in spinnaker:
``` json
"containers": [
        {
           ...,
          "volumeMounts": [
            {
              "mountPath": "/app/settings.json",
              "name": "demo-config",
              "subPath": "settings.json"
            }
          ]
        }
```

I'm expecting clouddriver to produce the following yaml
``` yaml
containers:
- volumeMounts:
  - name: demo-config
    mountPath: /app/settings.json
    subPath: settings.json
volumes:
- name: demo-config
  configMap:
    name: demo
```

but there's no property & mapping on the `KubernetesVolumeMount` class so the deployment looks like
``` yaml
containers:
- volumeMounts:
  - name: demo-config
    mountPath: /app/settings.json
volumes:
- name: demo-config
  configMap:
    name: demo
```

It would be nice to show this in the UI, but I'm happy to work with editing the json directly for the deployments that need this.

**Comment by jeckhart** [comment]
+1 on this, we use this the subpath to mount specific config files to a location.

**Comment by lwander** [comment]
Will be released in spinnaker 1.6
